lib/widgets/message_list.dart

```diff
@@ -1105,9 +1108,9 @@ class StreamMessageRecipientHeader extends StatelessWidget {
         ]));
 
     return GestureDetector(
-      onTap: () => Navigator.push(context,
-        MessageListPage.buildRoute(context: context,
-          narrow: TopicNarrow.ofMessage(message))),
+      onTap: !inTopicNarrow
+        ? () => Navigator.push(context, MessageListPage.buildRoute(context: context, narrow: TopicNarrow.ofMessage(message)))
+        : null,
```

Comment:
Please write a brief code comment explaining why we don't want a tap interaction here, and mentioning that this UI will change with #1039. (Refer to #1171 and linked discussion if you're not sure what to write here.)